spec/lib/subjects/postgresql_selection_spec.rb

```diff
@@ -35,6 +34,32 @@ def update_sms_priorities
             SetMemberSubject.all
           end
         end
+
+        context "with a training set and a real set with data" do
+          let(:sms_count) { 2 }
+          let(:training_set) { workflow.subject_sets.first }
+          let(:real_set) { workflow.subject_sets.last }
+
+          before do
+            training_set.update_column(
+              :metadata,
+              training_set.metadata.merge("training" => true)
+            )
+            create(:subject, project: workflow.project, uploader: uploader) do |subject|
+              create(:set_member_subject,
+                setup_subject_workflow_statuses: true,
+                subject: subject,
+                subject_set: real_set
+              )
+            end
+          end
+
+          it "should not include training subject sets in the results" do
+            result_ids = subject.select
```

Comment:
Why do you need the enumerator over just `[subject.id]`?

Reply:
I'm not sure i follow, `result_ids = subject.select` runs the selection code.

Reply:
I see. The words "select" and "subject" are used in the same code in completely different ways and I mistook the RSpec `subject` for an actual subject. Reading the same words over and over in different contexts and they soon lose all meaning...

Reply:
Gotcha. Yeah - i've never really liked the rspec subject syntax especially in our domain model. I can update the spec to be `let(:selector) { Subjects::PostgresqlSelection.new(workflow, user, opts) }` if you like but it'll have a bigger diff footprint.